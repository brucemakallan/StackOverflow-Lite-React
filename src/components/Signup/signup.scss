@import url('https://fonts.googleapis.com/css?family=Montserrat');
@import url('https://fonts.googleapis.com/css?family=Raleway:300');

$bodyFontFamily: 'Montserrat', sans-serif;
$headingFontFamily: 'Raleway', sans-serif;

body {
  font-family: $bodyFontFamily;
}
h1, h2, h3, h4, h5, h6 {
  font-family: $headingFontFamily;
}

.signup-box {
  display: flex;
  flex-flow: row wrap;
  padding: 30px 0;

  .signup-image, .signup-form {
    flex-grow: 1;
    width: 50%;
  }
  .signup-image {
    padding: 5px;
    img {
      padding: 5px;
      border: 1px solid #dddddd;
      width: 100%;
    }
  }
  .signup-form {
    padding: 30px;
  }
}

@media all and (max-width: 800px) {
  .signup-box {
    .signup-image, .signup-form {
      width: 100%;
    }
  }
}
