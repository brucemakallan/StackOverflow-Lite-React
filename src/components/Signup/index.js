import React from 'react';
import PropTypes from 'prop-types';
import './signup.scss';

const Signup = ({ onChange, onSubmit }) => {
	const createFormElement = (type, id, placeholder) => (
		<div className="form-group" key={id}>
			<input
				type={type}
				id={id}
				name={id}
				className="form-control"
				placeholder={placeholder}
				required
				onChange={onChange} />
		</div>
	);

	const formElements = [
		{
			type: 'text',
			id: 'fullName',
			placeholder: 'Full name'
		},
		{
			type: 'email',
			id: 'email',
			placeholder: 'email'
		},
		{
			type: 'password',
			id: 'password',
			placeholder: 'Password'
		},
		{
			type: 'password',
			id: 'retypePassword',
			placeholder: 'Retype password'
		}
	];

	return (
		<div className="container signup-box">
			<div className="signup-image">
				<img src="https://iviidev.info/downloads/images/questions.jpg" alt="poster" />
			</div>
			<form onSubmit={onSubmit} className="signup-form mt-4">
				<h3>Create a new account</h3>
				{formElements.map(el => createFormElement(el.type, el.id, el.placeholder))}
				<button type="submit"
					className="btn btn-primary">Register</button>
			</form>
		</div>
	);
};

Signup.propTypes = {
	onChange: PropTypes.func.isRequired,
	onSubmit: PropTypes.func.isRequired
};

export default Signup;
