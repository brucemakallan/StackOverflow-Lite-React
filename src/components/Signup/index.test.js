import React from 'react';
import { mount } from 'enzyme';
import Signup from '.';

describe('<Signup /> component', () => {
	test('renders correctly', () => {
		const wrapper = mount(<Signup onChange={jest.fn()} onSubmit={jest.fn()} />);
		expect(wrapper).toMatchSnapshot();
	});
});
