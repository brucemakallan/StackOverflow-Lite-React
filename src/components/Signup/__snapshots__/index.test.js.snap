// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> component renders correctly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Signup
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "onChange": [MockFunction],
      "onSubmit": [MockFunction],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="container signup-box"
      >
        <div
          class="signup-image"
        >
          <img
            alt="poster"
            src="https://iviidev.info/downloads/images/questions.jpg"
          />
        </div>
        <form
          class="signup-form mt-4"
        >
          <h3>
            Create a new account
          </h3>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              id="fullName"
              name="fullName"
              placeholder="Full name"
              required=""
              type="text"
            />
          </div>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              id="email"
              name="email"
              placeholder="email"
              required=""
              type="email"
            />
          </div>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              id="password"
              name="password"
              placeholder="Password"
              required=""
              type="password"
            />
          </div>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              id="retypePassword"
              name="retypePassword"
              placeholder="Retype password"
              required=""
              type="password"
            />
          </div>
          <button
            class="btn btn-primary"
            type="submit"
          >
            Register
          </button>
        </form>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="signup-image"
          >
            <img
              alt="poster"
              src="https://iviidev.info/downloads/images/questions.jpg"
            />
          </div>,
          <form
            className="signup-form mt-4"
            onSubmit={[MockFunction]}
          >
            <h3>
              Create a new account
            </h3>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="fullName"
                name="fullName"
                onChange={[MockFunction]}
                placeholder="Full name"
                required={true}
                type="text"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="email"
                name="email"
                onChange={[MockFunction]}
                placeholder="email"
                required={true}
                type="email"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="password"
                name="password"
                onChange={[MockFunction]}
                placeholder="Password"
                required={true}
                type="password"
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="retypePassword"
                name="retypePassword"
                onChange={[MockFunction]}
                placeholder="Retype password"
                required={true}
                type="password"
              />
            </div>
            <button
              className="btn btn-primary"
              type="submit"
            >
              Register
            </button>
          </form>,
        ],
        "className": "container signup-box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="signup-image"
          >
            <img
              alt="poster"
              src="https://iviidev.info/downloads/images/questions.jpg"
            />
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="poster"
              src="https://iviidev.info/downloads/images/questions.jpg"
            />,
            "className": "signup-image",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <img
                alt="poster"
                src="https://iviidev.info/downloads/images/questions.jpg"
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "poster",
                "src": "https://iviidev.info/downloads/images/questions.jpg",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": <form
            class="signup-form mt-4"
          >
            <h3>
              Create a new account
            </h3>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                id="fullName"
                name="fullName"
                placeholder="Full name"
                required=""
                type="text"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                id="email"
                name="email"
                placeholder="email"
                required=""
                type="email"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                id="password"
                name="password"
                placeholder="Password"
                required=""
                type="password"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                id="retypePassword"
                name="retypePassword"
                placeholder="Retype password"
                required=""
                type="password"
              />
            </div>
            <button
              class="btn btn-primary"
              type="submit"
            >
              Register
            </button>
          </form>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Create a new account
              </h3>,
              Array [
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    id="fullName"
                    name="fullName"
                    onChange={[MockFunction]}
                    placeholder="Full name"
                    required={true}
                    type="text"
                  />
                </div>,
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    id="email"
                    name="email"
                    onChange={[MockFunction]}
                    placeholder="email"
                    required={true}
                    type="email"
                  />
                </div>,
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    id="password"
                    name="password"
                    onChange={[MockFunction]}
                    placeholder="Password"
                    required={true}
                    type="password"
                  />
                </div>,
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    id="retypePassword"
                    name="retypePassword"
                    onChange={[MockFunction]}
                    placeholder="Retype password"
                    required={true}
                    type="password"
                  />
                </div>,
              ],
              <button
                className="btn btn-primary"
                type="submit"
              >
                Register
              </button>,
            ],
            "className": "signup-form mt-4",
            "onSubmit": [MockFunction],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <h3>
                Create a new account
              </h3>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Create a new account",
              },
              "ref": null,
              "rendered": Array [
                "Create a new account",
              ],
              "type": "h3",
            },
            Object {
              "instance": <div
                class="form-group"
              >
                <input
                  class="form-control"
                  id="fullName"
                  name="fullName"
                  placeholder="Full name"
                  required=""
                  type="text"
                />
              </div>,
              "key": "fullName",
              "nodeType": "host",
              "props": Object {
                "children": <input
                  className="form-control"
                  id="fullName"
                  name="fullName"
                  onChange={[MockFunction]}
                  placeholder="Full name"
                  required={true}
                  type="text"
                />,
                "className": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <input
                    class="form-control"
                    id="fullName"
                    name="fullName"
                    placeholder="Full name"
                    required=""
                    type="text"
                  />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "id": "fullName",
                    "name": "fullName",
                    "onChange": [MockFunction],
                    "placeholder": "Full name",
                    "required": true,
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": <div
                class="form-group"
              >
                <input
                  class="form-control"
                  id="email"
                  name="email"
                  placeholder="email"
                  required=""
                  type="email"
                />
              </div>,
              "key": "email",
              "nodeType": "host",
              "props": Object {
                "children": <input
                  className="form-control"
                  id="email"
                  name="email"
                  onChange={[MockFunction]}
                  placeholder="email"
                  required={true}
                  type="email"
                />,
                "className": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <input
                    class="form-control"
                    id="email"
                    name="email"
                    placeholder="email"
                    required=""
                    type="email"
                  />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "id": "email",
                    "name": "email",
                    "onChange": [MockFunction],
                    "placeholder": "email",
                    "required": true,
                    "type": "email",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": <div
                class="form-group"
              >
                <input
                  class="form-control"
                  id="password"
                  name="password"
                  placeholder="Password"
                  required=""
                  type="password"
                />
              </div>,
              "key": "password",
              "nodeType": "host",
              "props": Object {
                "children": <input
                  className="form-control"
                  id="password"
                  name="password"
                  onChange={[MockFunction]}
                  placeholder="Password"
                  required={true}
                  type="password"
                />,
                "className": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <input
                    class="form-control"
                    id="password"
                    name="password"
                    placeholder="Password"
                    required=""
                    type="password"
                  />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "id": "password",
                    "name": "password",
                    "onChange": [MockFunction],
                    "placeholder": "Password",
                    "required": true,
                    "type": "password",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": <div
                class="form-group"
              >
                <input
                  class="form-control"
                  id="retypePassword"
                  name="retypePassword"
                  placeholder="Retype password"
                  required=""
                  type="password"
                />
              </div>,
              "key": "retypePassword",
              "nodeType": "host",
              "props": Object {
                "children": <input
                  className="form-control"
                  id="retypePassword"
                  name="retypePassword"
                  onChange={[MockFunction]}
                  placeholder="Retype password"
                  required={true}
                  type="password"
                />,
                "className": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <input
                    class="form-control"
                    id="retypePassword"
                    name="retypePassword"
                    placeholder="Retype password"
                    required=""
                    type="password"
                  />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "id": "retypePassword",
                    "name": "retypePassword",
                    "onChange": [MockFunction],
                    "placeholder": "Retype password",
                    "required": true,
                    "type": "password",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": <button
                class="btn btn-primary"
                type="submit"
              >
                Register
              </button>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Register",
                "className": "btn btn-primary",
                "type": "submit",
              },
              "ref": null,
              "rendered": Array [
                "Register",
              ],
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "onChange": [MockFunction],
        "onSubmit": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="container signup-box"
        >
          <div
            class="signup-image"
          >
            <img
              alt="poster"
              src="https://iviidev.info/downloads/images/questions.jpg"
            />
          </div>
          <form
            class="signup-form mt-4"
          >
            <h3>
              Create a new account
            </h3>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                id="fullName"
                name="fullName"
                placeholder="Full name"
                required=""
                type="text"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                id="email"
                name="email"
                placeholder="email"
                required=""
                type="email"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                id="password"
                name="password"
                placeholder="Password"
                required=""
                type="password"
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control"
                id="retypePassword"
                name="retypePassword"
                placeholder="Retype password"
                required=""
                type="password"
              />
            </div>
            <button
              class="btn btn-primary"
              type="submit"
            >
              Register
            </button>
          </form>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="signup-image"
            >
              <img
                alt="poster"
                src="https://iviidev.info/downloads/images/questions.jpg"
              />
            </div>,
            <form
              className="signup-form mt-4"
              onSubmit={[MockFunction]}
            >
              <h3>
                Create a new account
              </h3>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  id="fullName"
                  name="fullName"
                  onChange={[MockFunction]}
                  placeholder="Full name"
                  required={true}
                  type="text"
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  id="email"
                  name="email"
                  onChange={[MockFunction]}
                  placeholder="email"
                  required={true}
                  type="email"
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  id="password"
                  name="password"
                  onChange={[MockFunction]}
                  placeholder="Password"
                  required={true}
                  type="password"
                />
              </div>
              <div
                className="form-group"
              >
                <input
                  className="form-control"
                  id="retypePassword"
                  name="retypePassword"
                  onChange={[MockFunction]}
                  placeholder="Retype password"
                  required={true}
                  type="password"
                />
              </div>
              <button
                className="btn btn-primary"
                type="submit"
              >
                Register
              </button>
            </form>,
          ],
          "className": "container signup-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="signup-image"
            >
              <img
                alt="poster"
                src="https://iviidev.info/downloads/images/questions.jpg"
              />
            </div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="poster"
                src="https://iviidev.info/downloads/images/questions.jpg"
              />,
              "className": "signup-image",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <img
                  alt="poster"
                  src="https://iviidev.info/downloads/images/questions.jpg"
                />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "poster",
                  "src": "https://iviidev.info/downloads/images/questions.jpg",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": <form
              class="signup-form mt-4"
            >
              <h3>
                Create a new account
              </h3>
              <div
                class="form-group"
              >
                <input
                  class="form-control"
                  id="fullName"
                  name="fullName"
                  placeholder="Full name"
                  required=""
                  type="text"
                />
              </div>
              <div
                class="form-group"
              >
                <input
                  class="form-control"
                  id="email"
                  name="email"
                  placeholder="email"
                  required=""
                  type="email"
                />
              </div>
              <div
                class="form-group"
              >
                <input
                  class="form-control"
                  id="password"
                  name="password"
                  placeholder="Password"
                  required=""
                  type="password"
                />
              </div>
              <div
                class="form-group"
              >
                <input
                  class="form-control"
                  id="retypePassword"
                  name="retypePassword"
                  placeholder="Retype password"
                  required=""
                  type="password"
                />
              </div>
              <button
                class="btn btn-primary"
                type="submit"
              >
                Register
              </button>
            </form>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Create a new account
                </h3>,
                Array [
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control"
                      id="fullName"
                      name="fullName"
                      onChange={[MockFunction]}
                      placeholder="Full name"
                      required={true}
                      type="text"
                    />
                  </div>,
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control"
                      id="email"
                      name="email"
                      onChange={[MockFunction]}
                      placeholder="email"
                      required={true}
                      type="email"
                    />
                  </div>,
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control"
                      id="password"
                      name="password"
                      onChange={[MockFunction]}
                      placeholder="Password"
                      required={true}
                      type="password"
                    />
                  </div>,
                  <div
                    className="form-group"
                  >
                    <input
                      className="form-control"
                      id="retypePassword"
                      name="retypePassword"
                      onChange={[MockFunction]}
                      placeholder="Retype password"
                      required={true}
                      type="password"
                    />
                  </div>,
                ],
                <button
                  className="btn btn-primary"
                  type="submit"
                >
                  Register
                </button>,
              ],
              "className": "signup-form mt-4",
              "onSubmit": [MockFunction],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <h3>
                  Create a new account
                </h3>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Create a new account",
                },
                "ref": null,
                "rendered": Array [
                  "Create a new account",
                ],
                "type": "h3",
              },
              Object {
                "instance": <div
                  class="form-group"
                >
                  <input
                    class="form-control"
                    id="fullName"
                    name="fullName"
                    placeholder="Full name"
                    required=""
                    type="text"
                  />
                </div>,
                "key": "fullName",
                "nodeType": "host",
                "props": Object {
                  "children": <input
                    className="form-control"
                    id="fullName"
                    name="fullName"
                    onChange={[MockFunction]}
                    placeholder="Full name"
                    required={true}
                    type="text"
                  />,
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <input
                      class="form-control"
                      id="fullName"
                      name="fullName"
                      placeholder="Full name"
                      required=""
                      type="text"
                    />,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "id": "fullName",
                      "name": "fullName",
                      "onChange": [MockFunction],
                      "placeholder": "Full name",
                      "required": true,
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": Array [
                      undefined,
                    ],
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": <div
                  class="form-group"
                >
                  <input
                    class="form-control"
                    id="email"
                    name="email"
                    placeholder="email"
                    required=""
                    type="email"
                  />
                </div>,
                "key": "email",
                "nodeType": "host",
                "props": Object {
                  "children": <input
                    className="form-control"
                    id="email"
                    name="email"
                    onChange={[MockFunction]}
                    placeholder="email"
                    required={true}
                    type="email"
                  />,
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <input
                      class="form-control"
                      id="email"
                      name="email"
                      placeholder="email"
                      required=""
                      type="email"
                    />,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "id": "email",
                      "name": "email",
                      "onChange": [MockFunction],
                      "placeholder": "email",
                      "required": true,
                      "type": "email",
                    },
                    "ref": null,
                    "rendered": Array [
                      undefined,
                    ],
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": <div
                  class="form-group"
                >
                  <input
                    class="form-control"
                    id="password"
                    name="password"
                    placeholder="Password"
                    required=""
                    type="password"
                  />
                </div>,
                "key": "password",
                "nodeType": "host",
                "props": Object {
                  "children": <input
                    className="form-control"
                    id="password"
                    name="password"
                    onChange={[MockFunction]}
                    placeholder="Password"
                    required={true}
                    type="password"
                  />,
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <input
                      class="form-control"
                      id="password"
                      name="password"
                      placeholder="Password"
                      required=""
                      type="password"
                    />,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "id": "password",
                      "name": "password",
                      "onChange": [MockFunction],
                      "placeholder": "Password",
                      "required": true,
                      "type": "password",
                    },
                    "ref": null,
                    "rendered": Array [
                      undefined,
                    ],
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": <div
                  class="form-group"
                >
                  <input
                    class="form-control"
                    id="retypePassword"
                    name="retypePassword"
                    placeholder="Retype password"
                    required=""
                    type="password"
                  />
                </div>,
                "key": "retypePassword",
                "nodeType": "host",
                "props": Object {
                  "children": <input
                    className="form-control"
                    id="retypePassword"
                    name="retypePassword"
                    onChange={[MockFunction]}
                    placeholder="Retype password"
                    required={true}
                    type="password"
                  />,
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <input
                      class="form-control"
                      id="retypePassword"
                      name="retypePassword"
                      placeholder="Retype password"
                      required=""
                      type="password"
                    />,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "id": "retypePassword",
                      "name": "retypePassword",
                      "onChange": [MockFunction],
                      "placeholder": "Retype password",
                      "required": true,
                      "type": "password",
                    },
                    "ref": null,
                    "rendered": Array [
                      undefined,
                    ],
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": <button
                  class="btn btn-primary"
                  type="submit"
                >
                  Register
                </button>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Register",
                  "className": "btn btn-primary",
                  "type": "submit",
                },
                "ref": null,
                "rendered": Array [
                  "Register",
                ],
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
